iwyu_tool.py

```diff
@@ -151,6 +151,12 @@ def main(compilation_db_path, source_files, verbose, formatter, iwyu_args):
                 print('WARNING: \'%s\' not found in compilation database.' %
                       source)
 
+    # Remove entries with directory belonging to any ignore_dirs
+    def is_ignored(dirpath):
+        return any(dirpath.startswith(os.path.realpath(i)) for i in ignore_dirs)
+
+    entries[:] = [e for e in entries if not is_ignored(e['directory'])]
```

Comment:
I think you can just rebind `entries` to the comprehension, rather than overwriting the array:

    entries = [e for e...]

Not sure which is better/faster here, but this seems clearer.